pyro/distributions/transforms/spline.py

```diff
@@ -0,0 +1,352 @@
+# This implementation is adapted in part from:
+# * https://github.com/tonyduan/normalizing-flows/blob/master/nf/flows.py; and,
+# * https://github.com/hmdolatabadi/LRS_NF/blob/master/nde/transforms/nonlinearities.py,
+# under the MIT license.
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.distributions import constraints
+
+from pyro.distributions.torch_transform import TransformModule
+from pyro.distributions.util import copy_docs_from
+
+# TODO: Move the spline functions to a pyro.numerics or torch.numerics module?
+# This would be a good place for e.g. numerical ODE solvers as well...
```

Comment:
I think our naming convention would be `pyro.ops.spline` and `pyro.ops.ode`. We tend to put all low level numerics in `pyro.ops`.

Reply:
If you do plan on moving these, cosider making them private for now, or moving them now.

Reply:
I've added an understand to the function names and will think about moving them in a future PR